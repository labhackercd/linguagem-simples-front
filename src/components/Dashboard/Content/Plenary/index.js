import React, { Component } from "react";
import { Grid } from '@material-ui/core';
import { makeStyles} from '@material-ui/core/styles';

import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Box from '@material-ui/core/Box';

import AgendaOfTheDayComponent from './AgendaOfTheDay/index'
import PlenaryVoting from './PlenaryVoting/index'


const useStyles = makeStyles((theme) => ({

  root: {
    flexGrow: 1,
    backgroundColor: '#007E5A',
    display: 'flex',
   
  },
  tabs: {
    borderRight: `2px solid ${theme.palette.divider}`,
  },
  tabIndicator: {
    backgroundColor: '#007E5A',
  },
}));

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box ml={1}>
          {children}
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`,
  };
}



export default function PlenaryPanel(props) {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);
  console.log(props)

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  if(props.sessionIdDadosAbertos === undefined || props.sessionIdDadosAbertos === null){
      return <div> Dados abertos não sincronizados </div>
  }else{
    return (
        <Grid container className={classes.body}>
            <Grid item xs={12}>
                <Tabs
                    orientation="horizontal"
                    value={value}
                    variant="fullWidth"
                    onChange={handleChange}
                    className={classes.tabs}
                    classes={{
                    indicator: classes.tabIndicator
                    }}
                >
                    <Tab label="Pauta do Dia" {...a11yProps(0)} />
                    <Tab label="Presença" {...a11yProps(1)} />
                    <Tab label="Votação" {...a11yProps(2)} />
                </Tabs>
            </Grid>
                <TabPanel value={value} index={0}>
                    <AgendaOfTheDayComponent sessionIdDadosAbertos={props.sessionIdDadosAbertos}></AgendaOfTheDayComponent>
                </TabPanel>
                <TabPanel value={value} index={1} >
                    oi 2
                </TabPanel>
                <TabPanel value={value} index={2}>
                    <PlenaryVoting sessionIdDadosAbertos={props.sessionIdDadosAbertos}></PlenaryVoting>
                </TabPanel>
        </Grid>
      )
  }


}
