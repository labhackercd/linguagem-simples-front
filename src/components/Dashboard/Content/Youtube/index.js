import React from 'react';
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography' 

import getYoutubeVideoUrl from './fetchYoutubeVideoUrl'

export default class Youtube extends React.Component {
  constructor(props){
    super(props);
    this.state = { 
        youtubeVideoId: null,
        sessionIdDadosAbertos: this.props.sessionIdDadosAbertos
    };
  }
  

  fetchYoutubeVideo = async term => {

    var data = null;
    //console.log(this.state.sessionIdDadosAbertos)
    if(this.state.sessionIdDadosAbertos !== undefined){
      //console.log(this.state.sessionIdDadosAbertos)
      data = await getYoutubeVideoUrl(this.state.sessionIdDadosAbertos);
      //This split is necessary because we need just the ID of video, since we need 
      //to use the tag embed in the url instead of the watch returned at API
      this.setState({youtubeVideoId:(data.split('?v=').pop())}) 
      this.setState({dataLoaded:true});
    }else{
      //Nothing to do
      //console.log("entrou no loop")
    }


  };

  componentDidMount(){
    this._isMounted = true;
    if(this._isMounted){
        console.log(this.props)
        this.fetchYoutubeVideo();
    }
  }

  render(){
    if(this.props.sessionIdDadosAbertos===null || this.state.youtubeVideoId === null){
      return (
        <Box width={1} height={1}>
          <Box display="flex" justifyContent="center" alignItems="center" width={"100%"} height={"100%"}>
            <Typography variant="h5" style={{color: "#C4C4C4"}}>Transmissão não disponível</Typography>
          </Box>
        </Box>
      )
    }

    return (
      <div
        className="video"
        id={"transmissaoYoutube"+this.state.youtubeVideoId}
        style={{
          position: "relative",
          paddingBottom: "56.25%" /* 16:9 */,
          paddingTop: 0,
          height: 0
        }}
      >
        <iframe
          style={{
            position: "absolute",
            top: 0,
            left: 0,
            width: "100%",
            height: "100%"
          }}
          src={`https://www.youtube.com/embed/${this.state.youtubeVideoId}`}
          frameBorder="0"
          title="Trasmissão da Sessão Plenária"
        />
      </div>
    );
  }
}
