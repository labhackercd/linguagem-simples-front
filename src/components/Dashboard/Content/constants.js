export let uiMessages = {
  clipboardCopySucess: "Link copiado com sucesso!",

}
