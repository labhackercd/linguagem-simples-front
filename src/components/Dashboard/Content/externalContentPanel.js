import React from 'react';
import { Grid, Typography, Button } from '@material-ui/core';
import { makeStyles, withStyles } from '@material-ui/core/styles';

import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Box from '@material-ui/core/Box';

import AgenciaCamaraContent from './AgenciaCamara/agenciaCamara'

const useStyles = makeStyles((theme) => ({
  body: {
    padding: '1rem',
    height: '100%',
    overflow: 'auto',
  },
  menu: {
    padding: '0.1rem',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  button: {
    margin: '0.1rem 0',
    width: '100%',
    height: '100% !important',
    color: 'white',
    '&:focus, &.Mui-focusVisible': {
      backgroundColor: '#007E5A',
      color: "white",
    },
  },
  content: {
    padding: '1rem',
  },
  root: {
    flexGrow: 1,
    backgroundColor: '#007E5A',
    display: 'flex',
    height: 224,
  },
  tabs: {
    borderRight: `2px solid ${theme.palette.divider}`,
  },
  tabIndicator: {
    backgroundColor: '#007E5A',
  },
}));

const MenuButton = withStyles((theme) => ({
  root: {
    color: "#666666",
    backgroundColor: "#F9F9F9",
    textTransform: 'none',
    '&:focus': {
      backgroundColor: '#007E5A',
      color: "white",
    },
  },
}))(Button);



function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          {children}
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`,
  };
}

export default function ExternalContentPanel() {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <Grid container className={classes.body}>
      <Grid item md={3} className={classes.menu}>
          <Tabs
            orientation="vertical"
            value={value}
            onChange={handleChange}
            aria-label="Vertical tabs example"
            variant="fullWidth"
            className={classes.tabs}
            classes={{
              indicator: classes.tabIndicator
            }}
          >
          
            <Tab label="Conteúdos Salvos" {...a11yProps(0)} />
            <Tab label="Agência Câmara" {...a11yProps(1)} />
            <Tab label="TV Câmara" {...a11yProps(2)} />
            <Tab label="Rádio Câmara" {...a11yProps(3)} />
            <Tab label="Twitter" {...a11yProps(4)} />
            <Tab label="Glossário" {...a11yProps(5)} />
          </Tabs>
      </Grid>
      <Grid item md={9} className={classes.content}>
        <Grid container>
            <Grid item xs={12}>Search</Grid>
            <Grid item xs={12}>
              <TabPanel value={value} index={0}>
              <AgenciaCamaraContent></AgenciaCamaraContent>
        
              </TabPanel>
              <TabPanel value={value} index={1}>
                Item Two
              </TabPanel>
              <TabPanel value={value} index={2}>
                Item Three
              </TabPanel>
              <TabPanel value={value} index={3}>
                Item Four
              </TabPanel>
              <TabPanel value={value} index={4}>
                Item Five
              </TabPanel>
              <TabPanel value={value} index={5}>
                Item Six
              </TabPanel>
            </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
