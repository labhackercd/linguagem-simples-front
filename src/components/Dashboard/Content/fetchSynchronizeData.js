import axiosInstance from './../../../auth/axiosApi'
import * as moment from 'moment'
import axios from 'axios'

export async function updateSession(sessionId,sessionIdDadosAbertos){
    const response = await axiosInstance.patch(('/sessions/'+sessionId+'/'), {id_session_dados_abertos: sessionIdDadosAbertos})
    if((parseInt(response.data.id_session_dados_abertos) === parseInt(sessionIdDadosAbertos))){
     //The session was succesfully updated
        return true;
    }else{
        return false
    }
}

export async function checkIfSessionsAlreadyExistsInSILEG(sessionDate) {
    const date = sessionDate
    //const date = (moment(sessionDate).format('YYYY-MM-DD'))
    const url =  "https://dadosabertos.camara.leg.br/api/v2/eventos?codTipoEvento=110&dataInicio="+date+"&dataFim="+date+"&ordem=ASC&ordenarPor=dataHoraInicio";

    const response = await axios.get(url)
    return  response.data;


}