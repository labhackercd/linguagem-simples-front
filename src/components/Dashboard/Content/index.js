import React from 'react';
import { Grid, Typography, Paper } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ExternalContentPanel from './externalContentPanel';
import Youtube from './Youtube/index'
import Button from '@material-ui/core/Button';
import VideoSnippets from './VideoSnippets/videoSnippets'
import {checkIfSessionsAlreadyExistsInSILEG,updateSession} from './fetchSynchronizeData'
import moment from 'moment'

const useStyles = makeStyles((theme) => ({
  body: {
    padding: '0 1.5rem'
  },
  header: {
    margin: '1rem 0',
    display: 'flex',
    justifyContent: 'space-between',
  },
  headerTitle: {
    display: 'flex',
    alignItems: 'flex-start',
    color: theme.palette.primary.main
  },
  headerMenu: {
    display: 'flex',
    alignItems: 'flex-end',
  },
  headerMenuItem: {
    padding: '0 1rem 0 1rem',
    alignSelf: 'flex-start'
  },
  column: {
    padding: "1rem",
  },
  firstRow: {
    margin: '0 0 0.5rem 0',
  },
  card: {
    backgroundColor: "white",
    height: 'auto',
    width: 'auto',
    margin: '1rem 0 0 0',
    minHeight: '35vh',
    minWidth: '32vh'
  },
  cardVideo: {
    backgroundColor: "white",
    height: '18vh',
    width: '32vh',
    margin: '1rem 0 0 0',
  },
  secondRow: {
    height: '16vh',
    backgroundColor: "white",
    borderRadius: '5px',
    margin: '0.5rem',
  },
  thirdRow: {
    backgroundColor: "white",
    height:'100%',
    borderRadius: '5px',
    margin: '0.5rem',
  },
  videoWrapper:{
    margin: '1rem 0 0 0',
  }
}));

export default function Content(props) {
  const sessionInfo = props.sessionInfo
  const sessionId = props.sessionID;

  if(sessionInfo!== undefined){
    var sessionIdDadosAbertos = sessionInfo.id_session_dados_abertos;
  }
  

  const classes = useStyles();

  async function handleSynchronize(){
    let sessionsScheduleDadosAbertos = await checkIfSessionsAlreadyExistsInSILEG((moment(sessionInfo.date).format('YYYY-MM-DD')));

    if(sessionsScheduleDadosAbertos.dados[0] ){
      // Session is registered at sileg, so update the information of dashboard
      const dashboardInfoUpdated = await updateSession(sessionId,sessionsScheduleDadosAbertos.dados[0].id)
  
      if(dashboardInfoUpdated){
        window.alert("Session ID atualizado com sucesso :D");
        window.location.reload(false);
      }else{
        window.alert("Falhou :(");
      }
    }
  }
  
  return (
		<Grid container className={classes.body}>
      <Grid container className={classes.header}>
        <Grid item className={classes.headerTitle}>
          <Typography variant="h3" className={classes.title}>Conteúdos</Typography>
        </Grid>
        <Grid item className={classes.headerMenu}>
          {!sessionIdDadosAbertos &&
            <Button onClick={handleSynchronize} className={classes.headerMenuItem}>Sincronizar</Button>
          }
            <Typography variant="h5" className={classes.headerMenuItem}> Ver Acompanhe </Typography>
        </Grid>
      </Grid>
      <Grid container className={classes.firstRow} spacing={2}>
          <Grid item md={7}>
            <Typography variant="h5"> Transmissão </Typography>
            <div className={classes.card}>{Youtube(sessionIdDadosAbertos)}</div>   
          </Grid>
          <Grid item md={5}>
            <Typography variant="h5"> Plenário </Typography>
            <Paper elevation={0} className={classes.card}> 
              
            </Paper>
          </Grid>
      </Grid>
      <Grid container className={classes.secondRow}>
        <VideoSnippets></VideoSnippets>
      </Grid>
      <Grid container className={classes.thirdRow}>
        <ExternalContentPanel></ExternalContentPanel>
      </Grid>
		</Grid>
  )
}
