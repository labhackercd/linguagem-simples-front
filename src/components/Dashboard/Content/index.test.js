import React from 'react';
import { render, screen } from '@testing-library/react';
import Content from './index';
import {shallow} from "enzyme/build";

it("should render the content section", () => {
    const component = shallow(<Content/>);
    expect(component).toMatchSnapshot();
});
