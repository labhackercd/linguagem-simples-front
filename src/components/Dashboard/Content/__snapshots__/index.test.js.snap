// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the content section 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-body-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-header-2"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-headerTitle-3"
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h3"
      >
        Conteúdos
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-headerMenu-4"
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-headerMenuItem-5"
        variant="h5"
      >
         Sincronizar 
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-headerMenuItem-5"
        variant="h5"
      >
         Ver Acompanhe 
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-firstRow-7"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
         Transmissão 
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-card-8"
        elevation={0}
      >
        <img
          alt="video placeholder"
          src="../../img/video_placeholder.png"
          style={
            Object {
              "maxHeight": "100%",
              "maxWidth": "100%",
            }
          }
        />
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
         Plenário 
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-card-8"
        elevation={0}
      >
         
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-secondRow-9"
    container={true}
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "margin": "1rem 0 1rem 1rem",
        }
      }
      variant="h5"
    >
       Trechos 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-thirdRow-10"
    container={true}
  >
    <ExternalContentPanel />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
